docs/content/users/configuration/config_yaml.md

```diff
@@ -1,76 +1,561 @@
+
+=== "config.yaml"
+
+    ```yaml
+    php_version: "8.1"
+    ```
+=== "`ddev config`"
+
+    ```shell
+    ddev config --php-version 8.1
+    ```
+
+    Run `ddev help config` to see all the available config arguments.
+
+### Environmental Overrides
+
+You can override the per-project `config.yaml` with files named `config.*.yaml`.
+
+For example, many teams use `config.local.yaml` for configuration that’s specific to one environment, and not checked into the team’s default `config.yaml`.
+
+---
+
+## `additional_fqdns`
+
+An array of [extra fully-qualified domain names](../extend/additional-hostnames.md) to be used for a project.
+
+| Type | Default | Usage
+| -- | -- | --
+| :octicons-file-directory-16: project | `[]` |
+
+Example: `additional_fqdns: ["example.com", "sub1.example.com"]` would provide HTTP and HTTPS URLs for `example.com` and `sub1.example.com`.
+
+!!!warning
+    Take care with this; it adds items to your `/etc/hosts` file which can cause confusion.
+
+## `additional_hostnames`
+
+An array of [extra hostnames](../extend/additional-hostnames.md) to be used for a project.
+
+| Type | Default | Usage
+| -- | -- | --
+| :octicons-file-directory-16: project | `[]` |
+
+Example: `additional_hostnames: ["somename", "someothername", "*.thirdname"]` would provide HTTP and HTTPS URLs for `somename.ddev.site`, `someothername.ddev.site`, and `one.thirdname.ddev.site` + `two.thirdname.ddev.site`.
+
+The wildcard (`*`) setting only works if you’re using DNS to resolve hostnames (default) and connected to the internet.
+
+## `composer_root`
+
+The relative path, from the project root, to the directory containing `composer.json`. (This is where all Composer-related commands are executed.)
+
+| Type | Default | Usage
+| -- | -- | --
+| :octicons-file-directory-16: project | |
+
+## `composer_version`
+
+Composer version for the web container and the `ddev composer` command.
+
+| Type | Default | Usage
+| -- | -- | --
+| :octicons-file-directory-16: project | `2` | Can be `2`, `1`, or empty (`""`) for latest major version at container build time.<br><br>Can also be a minor version like `2.2` for the latest release of that branch, an explicit version like `1.0.22`, or a keyword like `stable`, `preview` or `snapshot`. See Composer documentation.
+
+## `database`
+
+The type and version of the database engine the project should use.
+
+| Type | Default | Usage
+| -- | -- | --
+| :octicons-file-directory-16: project | MariaDB 10.3 | Can be MariaDB 5.5–10.7, MySQL 5.5–8.0, or PostgreSQL 9–14.<br>See [Database Server Types](../extend/database_types.md) for examples and caveats.
+
+!!!note
+    DDEV v1.18 and earlier used `mariadb_version` and `mysql_version`, which are now automatically converted to the `database` format.
+
+## `dbimage_extra_packages`
+
+Extra Debian packages for the project’s database container.
+
+| Type | Default | Usage
+| -- | -- | --
+| :octicons-file-directory-16: project | `[]` |
```

Comment:
> This is great communication in rendered page (the images). I wish it could be meaningful in the source though.

Same. In VuePress world we could introduce custom elements like `<icon kind="globe"/>` or whatever. Not sure if there’s a better way here.

> I'm thinking that "empty" might be more descriptive than []...

I thought `[]` indicated an empty _array_ whereas the word “empty” is less specific about the expected format. But this isn’t a hill I’d like to die on.